### PR TITLE
chore: update horovod version 

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
This new version of horovod correctly increments `iterations` in tf.keras optimizers
and support gradient aggregation for Tensorflow 2.4.

## Test Plan
Improved gradient aggregation tests, also going to run full suite of e2e GPU 
before merging.

### CODE REVIEW

**Comment by aaron276h** [comment]
E2E tests passed: https://app.circleci.com/pipelines/github/determined-ai/determined/6291/workflows/b32292cd-8956-4eb3-b059-8a9df36a265b

File: e2e_tests/tests/fixtures/tf_keras_grad_aggregation/grad_aggregation_test.py [hunk 1]
```diff
@@ -11,16 +13,26 @@
...
+
 class CustomOptimizer(optimizer_v2.OptimizerV2):
     def get_config(self) -> Any:
         config = super(CustomOptimizer, self).get_config()
         return config
 
+    def _create_slots(self, var_list: Any) -> None:
+        pass
+
+    def _resource_apply_dense(self, grad: Any, var: Any, apply_state: Any = None) -> Any:
```
Comment (armandmcqueen) [inline e2e_tests/tests/fixtures/tf_keras_grad_aggregation/grad_aggregation_test.py:27]: Can we use a more descriptive type annotation here?

**Review by armandmcqueen — APPROVED** [review]
LGTM

File: e2e_tests/tests/fixtures/tf_keras_grad_aggregation/grad_aggregation_test.py (hunk 1 above)
Comment (aaron276h) [reply e2e_tests/tests/fixtures/tf_keras_grad_aggregation/grad_aggregation_test.py:27]: fairly sure mypy interprets `tf.Tensors` and `tf.Variable` as `Any`
